code/game/machinery/firealarm.dm

```diff
@@ -28,15 +28,12 @@
 	resistance_flags = FIRE_PROOF
 
 
-/obj/machinery/firealarm/New(loc, dir, building)
+/obj/machinery/firealarm/Construct(mob/living/user, ndir)
 	..()
-	if(dir)
-		src.setDir(dir)
-	if(building)
-		buildstage = 0
-		panel_open = 1
-		pixel_x = (dir & 3)? 0 : (dir == 4 ? -24 : 24)
-		pixel_y = (dir & 3)? (dir ==1 ? -24 : 24) : 0
+	buildstage = 0
+	panel_open = 1
+	pixel_x = (dir & 3)? 0 : (dir == 4 ? -24 : 24)

```

Comment:
I don't like the ```?``` hugging its operands, that includes the parenthesis.

also these dir constants should be actual values, for example: 4 -> ```EAST```, 3 -> ```NORTH|SOUTH``` (probably worth a ```#define NORTHSOUTH 3``` so the define doesn't cause the ```|``` to happen every time it's used)